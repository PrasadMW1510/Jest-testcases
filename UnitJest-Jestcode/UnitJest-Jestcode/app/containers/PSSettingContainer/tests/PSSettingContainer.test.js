import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { fromJS } from 'immutable';
import React from 'react';

import { PSSettingContainer } from '../index';

describe('<PSSettingContainer />', () => {
  let wrapper = null;

  let mockCohortObj = null;
  let mockPSSettingsContainerRequest = null;
  let mockPSSettingsSave = null;
  let mockEnrollmentCount = null;
  let mockPSSettingData = null;

  beforeEach(() => {
    mockCohortObj = fromJS({
      cohortType: 'mockCohType',
    });

    mockPSSettingsContainerRequest = jest.fn();
    mockPSSettingsSave = jest.fn();
    mockEnrollmentCount = 10;
    mockPSSettingData = fromJS({
      loading: false,
      settings: {},
    });

    wrapper = shallow(
      <PSSettingContainer
        cohortObj={mockCohortObj}
        psSettingsContainerRequest={mockPSSettingsContainerRequest}
        psSettingsSave={mockPSSettingsSave}
        enrollmentCount={mockEnrollmentCount}
        psSettingData={mockPSSettingData}
      />
    );
  });

  it('Expect to render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Expect psSettingsContainerRequest to have been called', () => {
    expect(mockPSSettingsContainerRequest).toHaveBeenCalled();
  });

  it('handleSave to have called psSettingsSave', () => {
    const mockUpdatedSettings = 'mockUpdatedSettings';
    wrapper.instance().handleSave(mockUpdatedSettings);

    expect(mockPSSettingsSave).toHaveBeenCalledWith(mockUpdatedSettings);
  });
});
