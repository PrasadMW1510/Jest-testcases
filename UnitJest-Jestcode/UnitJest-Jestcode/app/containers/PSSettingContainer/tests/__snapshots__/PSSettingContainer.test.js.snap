// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PSSettingContainer /> Expect to render correctly 1`] = `
<PSSettings
  cohortObj={
    Immutable.Map {
      cohortType: "mockCohType",
    }
  }
  enrollmentCount={10}
  handleSave={[Function]}
  isLoading={false}
  settings={
    Immutable.Map {
    }
  }
  testAssignmentData={
    Immutable.Map {
      loading: false,
      settings: Immutable.Map {
      },
    }
  }
/>
`;
